src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,242 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { useCallback, useRef } from 'react';
+import { NumberInputIncrement, NumberInputProps } from './types';
+import { merge } from '@utilities/merge';
+import { IconCheckMark16, IconExclamationMarkTriangle16, IconMinus16, IconPlus16 } from '@foundation/Icon';
+import { useMemoizedId } from '@hooks/useMemoizedId';
+import { FOCUS_WITHIN_STYLE } from '@utilities/focusStyle';
+import { Validation, validationClassMap } from '@utilities/validation';
+import { KeyboardEvent } from '@react-types/shared';
+
+const INCREMENT_KEYS = ['ArrowUp', 'ArrowRight'];
+const DECREMENT_KEYS = ['ArrowDown', 'ArrowLeft'];
+
+export const NumberInput = ({
+    id: propId,
+    size,
+    disabled = false,
+    required,
+    readOnly,
+    status = Validation.Default,
+    decorator,
+    incrementable,
+    placeholder,
+    stepInterval = 10,
+    title = 'Fondue Number Input',
```

Comment:
```suggestion
    title,
```
No default `title` otherwise on hover all number input would have it displayed